manifests/install.pp

```diff
@@ -30,16 +30,27 @@
     creates      => $version_file,
     proxy_server => $winlogbeat::proxy_address,
   }
-
-  exec { "unzip ${filename}":
-    command => "\$sh=New-Object -COM Shell.Application;\$sh.namespace((Convert-Path '${winlogbeat::install_dir}')).Copyhere(\$sh.namespace((Convert-Path '${zip_file}')).items(), 16)",
-    creates => $version_file,
-    require => [
-      File[$winlogbeat::install_dir],
-      Archive[$zip_file],
-    ],
+  # use Expand-Archive cmdlet if available (pwsh >= 5), as server core installations
+  # run into issues using the original method
+  if $facts['psversiontable']['psversion']['major'] >= 5 {
```

Comment:
this only works if the fact is present. using the fact() method from stdlib is probably the better approach. can you please add a unit test for this fact as well?

Reply:
done!